packages/vike-react-styled-jsx/onAfterRenderHtml.tsx

```diff
@@ -0,0 +1,26 @@
+export { onAfterRenderHtml }
+
+import React from 'react'
+import { useConfig } from 'vike-react/useConfig'
+import type { PageContext } from 'vike/types'
+
+function onAfterRenderHtml(pageContext: PageContext) {
+  const config = useConfig()
+  const registry = pageContext.styledJsxRegistry
+
+  if (registry) {
+    const nonce = pageContext.config.styledJsx?.nonce
+    const styles = registry.styles({ nonce })
+
+    config({
+      Head: (
+        <>
+          {nonce ? <meta property="csp-nonce" content={nonce} /> : ''}
```

Comment:
I think it's the user who should inject that tag, not `vike-react-styled-jsx`. See:

> Your CSP policy must share the same nonce as well (the header nonce needs to match the html nonce and remain unpredictable). Content-Security-Policy: default-src 'self'; style-src 'self' 'nonce-N2M0MDhkN2EtMmRkYi00MTExLWFhM2YtNDhkNTc4NGJhMjA3';

Instead, let's add a note in the readme above that the user should have that tag (he most likely already has if he's using a nonce).